scipy/stats/tests/test_distributions.py

```diff
@@ -2805,6 +2805,20 @@ def test_rayleigh_accuracy():
     assert_almost_equal(p, 9.0, decimal=15)
 
 
+def test_genextreme_give_no_warnings():
+    """regression test for gh-6219"""
+
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+
+        p = stats.genextreme.cdf(.5, 0)
+        p = stats.genextreme.pdf(.5, 0)
+        p = stats.genextreme.ppf(.5, 0)
+        p = stats.genextreme.logpdf(-np.inf, 0.0)
+        number_of_warnings_thrown = len(w)
+        assert_equal(number_of_warnings_thrown, 0)
```

Comment:
Wow, fancy. TIL that `warnings.catch_warnings()` has length, which is the number of warnings thrown. I guess a less sophisticated way is to use `warnings.simplefilter('errors')` within a context manager so that a warning gets converted to an error and a test fails in a usual way, but OK, let's keep this fanciness.